functions/ecs-shutdown-services/enable.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# What SaaS Boost installation are we working on?
+SAAS_BOOST_ENV=$1
+if [ -z "$SAAS_BOOST_ENV" ]; then
+    read -p "Enter your AWS SaaS Boost Environment label: " SAAS_BOOST_ENV
+    if [ -z "$SAAS_BOOST_ENV" ]; then
+        echo "You must enter a AWS SaaS Boost Environment label to continue. Exiting."
+        exit 1
+    fi
+fi
+# Can we confirm that this AWS CLI is connected to that SaaS Boost environment?
+MY_AWS_REGION=$(aws configure list | grep region | awk '{print $2}')
+SB_ENV=$(aws --region ${MY_AWS_REGION} ssm get-parameter --name /saas-boost/${SAAS_BOOST_ENV}/SAAS_BOOST_ENVIRONMENT --query "Parameter.Value" --output text)
+if [ "${SB_ENV}" != "${SAAS_BOOST_ENV}" ]; then
+    echo "Can't find SaaS Boost environment $SAAS_BOOST_ENV in region $MY_AWS_REGION. Double-check the current AWS CLI profile and region."
+    exit 1
+fi
+
+# Has the EcsShutdownServices Lambda function been setup?
+LAMBDA_FX="sb-${SAAS_BOOST_ENV}-ecs-shutdown-services"
+LAMBDA_ARN=$(aws lambda get-function --function-name "${LAMBDA_FX}" --query "Configuration.FunctionArn" --output text)
+if [ $? != 0 ]; then
+    echo "Can't find the EcsShutdownServices Lambda function in this SaaS Boost environment."
+    exit 1
+fi
+
+RULE="sb-${SAAS_BOOST_ENV}-ecs-shutdown-services"
+EXISTING_SCHEDULE=$(aws events describe-rule --name "${RULE}" --query "ScheduleExpression" --output text)
+if [ $? == 0 ] && [ ! -z "${EXISTING_SCHEDULE}" ]; then
+    read -p "Reuse the existing schedule expression ${EXISTING_SCHEDULE}? [Y/N] " REUSE_SCHEDULE
+fi
+if ! [[ $REUSE_SCHEDULE =~ ^[Yy] ]]; then
+    # Get a cron schedule to invoke our Lambda on
+    echo "Enter an EventBridge cron schedule expression (https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-create-rule-schedule.html#eb-cron-expressions)."
+    read -p "[Press enter for default schedule of 12 midnight daily]: " SCHEDULE
+    if [ -z "$SCHEDULE" ]; then
+        SCHEDULE="cron(0 0 * * ? *)"
+    fi
+else
+    SCHEDULE="${EXISTING_SCHEDULE}"
+fi
+
+# Simple pattern check -- doesn't test for EventBridge restriction on
+# day-of-month and day-of-week not both being asterisks
+CRON_REGEX="^cron\(.+ .+ .+ .+ .+ .+\)$"
+if ! [[ $SCHEDULE =~ $CRON_REGEX ]]; then
+    echo "Schedule cron expression ${SCHEDULE} is not valid."
+    exit 1
+fi
```

Comment:
What if I don't want to set up a schedule for the EventBridge rule? Should there be a way to just use this scripts to fire the event for the disable/enable services lambdas?

Reply:
Why wouldn't you just invoke the Lambda if you didn't want it on a schedule? You don't need a script for that.
`aws lambda invoke --function-name sb-${SAAS_BOOST_ENV}-ecs-shutdown-services response.json`

Reply:
My understanding was that this PR is targeting ease of use improvements for developers of SaaS Boost. I personally wouldn't see having to manually invoke a lambda function as a particularly usable way to shutdown tenant services, especially if there is already a script that is basically doing the same thing.

I'll leave this up to you. If doing it without a schedule causes the script to fail, maybe some error message in the script would be useful:
``"Schedule cron expression ${SCHEDULE} is not valid. To just disable tenant services, run `aws lambda invoke --function-name sb-${SAAS_BOOST_ENV}-ecs-shutdown-services response.json`"``